Content.Server/GameObjects/Components/Chemistry/ReagentDispenserComponent.cs

```diff
@@ -142,6 +146,26 @@ private void OnUiReceiveMessage(ServerBoundUserInterfaceMessage obj)
             ClickSound();
         }
 
+        /// <summary>
+        /// Checks whether the player entity is able to use the chem dispenser.
+        /// </summary>
+        /// <param name="playerEntity">The player entity.</param>
+        /// <returns>Returns true if the entity can use the dispenser, and false if it cannot.</returns>
+        private bool PlayerCanUseDispenser(IEntity playerEntity)
+        {
+            //Need player entity to check if they are still able to use the dispenser
+            if (playerEntity == null) 
+                return false;
+            //Get species component that has damage state
+            if (!playerEntity.TryGetComponent(out SpeciesComponent species)) 
+                return false;
+            //Check if player can interact in their current state
+            if (!species.CurrentDamageState.CanInteract() || !species.CurrentDamageState.CanUse())
```

Comment:
You should not be checking the species component directly. You should be checking via `ActionBlockerSystem` instead.

Reply:
Ah okay. I switched it to use that instead. Should've looked more closely at how other code was checking this.